### ISSUE TITLE
[Bug] Landing Page Experiment causes 404 error when visiting media attachment pages

### ISSUE THREAD

**Issue by zeinnicholas** [elementor/elementor#15943]
## Prerequisites

- [x] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [x] The issue still exists against the latest stable version of Elementor.

## Description

When activating the **Landing Page Experiment**, users are reporting not being able to visit Media Attachment pages, causing a 404 error. 

The issue was reported happening using only Elementor (3.3.1), the Hello Elementor theme (2.4.1), and also the Twenty-twenty-one theme (1.4).

Deactivating the experiment seems to fix the issue.

This might be related to this other issue:
https://github.com/elementor/elementor/issues/15903

This was reported in the **Elementor Leaders** Slack Channel as well as in the WordPress Forum:
https://wordpress.org/support/topic/media-attachment-pages-show-404-not-found/

![attachment-pages](https://user-images.githubusercontent.com/26891676/128873034-b7e4d31c-4b76-4ff7-84a3-a0ba2bb7c1ee.gif)

## Steps to reproduce

1. Install and activate Elementor 3.3.1 using any theme;
2. Make sure the Landing Page Experiment is active;
3. Go to the Media Library and upload an image;
4. Click to see the Attachment Page;
5. A "Nothing found" page will appear with a 404 error in the console.

## Temporary Fix

1. Deactivate the Landing Page Experiment.

## Isolating the problem
- [x] This bug happens with only Elementor plugin active (and Elementor Pro).
- [x] This bug happens with a default WordPress theme active.
- [x] I can reproduce this bug consistently using the steps above.

## Environment

<details>
<summary>System Info</summary>
```
					
== Server Environment ==
	Operating System: Linux
	Software: nginx/1.19.3
	MySQL version: mariadb.org binary distribution v10.4.21
	PHP Version: 7.4.21
	PHP Max Input Vars: 1000
	PHP Max Post Size: 256M
	GD Installed: Yes
	ZIP Installed: Yes
	Write Permissions: All right
	Elementor Library: Connected

== WordPress Environment ==
	Version: 5.8
	Site URL: [redacted]
	Home URL: [redacted]
	WP Multisite: No
	Max Upload Size: 256 MB
	Memory limit: 256M
	Permalink Structure: /%postname%/
	Language: en-US
	Timezone: 0
	Debug Mode: Inactive

== Theme ==
	Name: Twenty Twenty-One
	Version: 1.4
	Author: the WordPress team
	Child Theme: No

== User ==
	Role: administrator
	WP Profile lang: en_US
	User Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/92.0.4515.131 Safari/537.36

== Active Plugins ==
	Elementor
		Version: 3.3.1
		Author: Elementor.com


== Elementor - Compatibility Tag ==
	


== Elements Usage ==
	


== Log ==
	

				
```
</details>

**Comment by krishneup** [comment]
@zeinnicholas thank you so much.

**Comment by shilo-ey** [comment]
This issue has been resolved in Elementor v3.4.3

Feel free to update

Thanks!

**Comment by cirkut** [comment]
This update broke all landing pages across multiple sites for me. Rolling back to 3.4.2 restores them.

**Comment by zeinnicholas** [comment]
> @shilo-ey This update broke all landing pages across multiple sites for me. Landing pages result in 404 errors.
> 
> Rolling back to 3.4.2 restores them.

Before confirming this is a real bug, can you try to reset your permalinks? Go to your Dashboard > Settings > Permalinks and hit "Save Changes" twice. Then make sure you clear all types of cache, your browser cookies, and also DNS cache ([How to clear the DNS cache on your computer](https://www.a2hosting.com.br/kb/getting-started-guide/internet-and-networking/clearing-the-dns-cache-on-your-computer)).

**Comment by cirkut** [comment]
@zeinnicholas This was the first step for me prior to commenting here initially. Permalinks changed and then changed back, caching plugin disabled, site/server cache cleared, cloudflare development mode enabled. 3.4.3 broke simple sites that are using landing pages and more complex ones.

DNS cache will have nothing to do with this issue since that's for domain resolution and no hosting or DNS changes were made. Simply rolling back and forth between 3.4.2 and 3.4.3 breaks and unbreaks landing pages without any of the above steps as well.

**Comment by zeinnicholas** [comment]
@cirkut Awesome, so this definitely needs to be addressed. I would advise you to open a new issue and reference this one so they can address it properly. ;)

**Comment by cirkut** [comment]
@zeinnicholas Good point, I've opened the new issue but had initially commented here to potentially quickly get this on the dev team's radar since it's what I would consider a relatively big breaking change that should be addressed ASAP.

Thanks for keeping me in check ;)

**Comment by shilo-ey** [comment]
Thank you @cirkut @zeinnicholas !

We will review it and release a patch if needed.

Thank you for raising the issue.
